Patch to fix a few warnings.
Fix for strtok_r is for "expected ‘char * __restrict__’ but argument is of type
‘const char *’" warning.
Second is to fix "cast to pointer from integer of different size" warning.
==========================================================================
diff --git a/src/va.c b/src/va.c
index bc469a0..09a7a3c 100644
--- a/src/va.c
+++ b/src/va.c
@@ -141,7 +141,7 @@ static VAStatus va_openDriver(VADisplay dpy, char *driver_name)
     }
 
     search_path = strdup((const char *)search_path);
-    driver_dir = strtok_r((const char *)search_path, ":", &saveptr);
+    driver_dir = strtok_r(search_path, ":", &saveptr);
     while(driver_dir)
     {
         void *handle = NULL;
diff --git a/test/putsurface/putsurface.c b/test/putsurface/putsurface.c
index 335368a..638bee3 100644
--- a/test/putsurface/putsurface.c
+++ b/test/putsurface/putsurface.c
@@ -241,7 +241,7 @@ static int putsurface_thread(void *data)
         while (surface_id == VA_INVALID_SURFACE)
             surface_id = get_next_free_surface(&index);
 
-        if (verbose) printf("Thread %x Display surface 0x%p,\n", (unsigned int)win, (void *)surface_id);
+        if (verbose) printf("Thread %x Display surface 0x%p,\n", (unsigned int)win, &surface_id);
 
         upload_surface(va_dpy, surface_id, box_width, row_shift, display_field);
 
